src/navigation/path.cpp

```diff
@@ -183,6 +241,45 @@ glm::dquat Path::easedSlerpRotation(double t) const {
     return glm::slerp(_start.rotation(), _end.rotation(), tScaled);
 }
 
+glm::dquat Path::linearPathRotation(double t) const {
+    const double tHalf = 0.5;
+
+    const glm::dvec3 endNodePos = _end.node()->worldPosition();
+    const glm::dvec3 endUp = _end.rotation() * glm::dvec3(0.0, 1.0, 0.0);
+
+    if (t < tHalf) {
+        // Interpolate to look at target
+        const glm::dvec3 halfWayPosition = _curve->positionAt(tHalf);
+        const glm::dquat q = ghoul::lookAtQuaternion(halfWayPosition, endNodePos, endUp);
+
+        const double tScaled = ghoul::sineEaseInOut(t / tHalf);
+        return glm::slerp(_start.rotation(), q, tScaled);
+    }
+
+    // This distance is guaranteed to be strictly decreasing for linear paths
+    const double distanceToEnd = glm::distance(_prevPose.position, _end.position());
+
+    double closingUpDistance = 10.0 * _end.validBoundingSphere();
```

Comment:
Maybe just a comment here that specifies that 10, 2. and 0.2 are just randomly picked constants